things/postgres/channels.go

```diff
@@ -314,29 +314,33 @@ func (cr channelRepository) Connect(ctx context.Context, owner string, chIDs, th
 	return nil
 }
 
-func (cr channelRepository) Disconnect(ctx context.Context, owner, chanID, thingID string) error {
+func (cr channelRepository) Disconnect(ctx context.Context, owner string, chIDs, thIDs []string) error {
 	q := `DELETE FROM connections
 	      WHERE channel_id = :channel AND channel_owner = :owner
 	      AND thing_id = :thing AND thing_owner = :owner`
 
-	conn := dbConnection{
-		Channel: chanID,
-		Thing:   thingID,
-		Owner:   owner,
-	}
+	for _, chID := range chIDs {
+		for _, thID := range thIDs {
+			dbco := dbConnection{
+				Channel: chID,
+				Thing:   thID,
+				Owner:   owner,
+			}
 
-	res, err := cr.db.NamedExecContext(ctx, q, conn)
-	if err != nil {
-		return errors.Wrap(things.ErrDisconnect, err)
-	}
+			res, err := cr.db.NamedExecContext(ctx, q, dbco)
+			if err != nil {
```

Comment:
Do we need to roll back transactions and return in case of error, as we do on [connect](https://github.com/mainflux/mainflux/blob/2f4abd8b228f6c21d3cde2bcf4223bb6c5951fa6/things/postgres/channels.go#L293)?

Reply:
Yes, we need transaction roll back in case of any error; since we are executing multiple queries that are dependent on each other. I missed that point. I will update the PR.